templates/analysis.yml

```diff
@@ -0,0 +1,123 @@
+# SonarCloud configuration for Maven builds expects JDK 11
+#
+# Functionality Caveat: Angular builds not successfully tested
+# ---------------------
+# All code blocks using the angular flag have not been successfully tested.  
+# The code not been removed for three reasons:
+# 1) it functions without a template, and so provides a central record of how
+#    the Angular build tasks function
+# 2) we hope to revisit and debug the errors that prevent the template from
+#    functioning as expected.  As it stands, it appears that the Angular tasks
+#    do not function when a job checks out multiple repositories for some reason
+# 3) We believe that the code in this template is correct, but the error appears
+#    to exist on the side of the calling YAML file and associated setup, therefore
+#    a savvy user may be able to utilize the functionality even if we were not
+#    successful in doing so.
+# Therefore, the code is included with the caveat of not being successfully tested
+
+parameters:
+- name: microservice  # <-- e.g., gateway, account-microservice
+  type: string
+- name: coveragePath  # <-- may override the coverage path if desired
+  type: string
+  default: ''
+- name: workDir       # <-- will need to override with multi-repo subdirectory
+  type: string        #     e.g., ./project-registry-tracking-microservice/
+- name: angular       # <-- Not successfully tested; see note above
+  type: boolean       #     (default is a Maven build when angular = false
+  default: false      #      the Angular build steps will substitute when angular = true)
+- name: sonarKeyPrefix
+  type: string
+- name: sonarNamePrefix
+  type: string
+- name: sonarOrg
+  type: string
+- name: sonarUrlPrefix
+  type: string
+- name: discordChannel
+  type: string
+- name: discordKey
+  type: string
+
+steps:
+- ${{ if eq(parameters.angular, true) }}: # <-- Angular build not successfully tested; see note above
+  - task: SonarCloudPrepare@1 # <-- See external requirements note above
+    inputs:
+      SonarCloud: 'SonarCloud'
+      organization: ${{ parameters.sonarOrg }} 
+      scannerMode: 'CLI'
+      configMode: 'manual'
+      cliProjectKey: ${{ parameters.sonarKeyPrefix }}${{ parameters.microservice }}
+      cliSources: $(System.DefaultWorkingDirectory)/${{ parameters.workDir }}
+    displayName: Prepare Analysis for ${{ parameters.microservice }} --> Task
+  
+  - script: cd $(System.DefaultWorkingDirectory)/${{ parameters.workDir }} && pwd && ls
+    displayName: Show path and contents
+
+  - script: cd $(System.DefaultWorkingDirectory)/${{ parameters.workDir }} && npm ci
+    displayName: Install dependencies for ${{ parameters.microservice }} --> Task
+
+  - script: cd $(System.DefaultWorkingDirectory)/${{ parameters.workDir }} && npm run build
```

Comment:
this "cd __ &&" business is exactly what the `workingDirectory` option for the script step is for, so use that instead.
also, you usually don't need to qualify things with System.DefaultWorkingDirectory, unless you really need an absolute path for some reason. side note, i'm pretty sure these examples would fail as-is if there were spaces in the DefaultWorkingDirectory.

Reply:
Updated per your suggestion.